sdks/python/apache_beam/io/localfilesystem.py

```diff
@@ -136,10 +137,17 @@ def _path_open(
       mode,
       mime_type='application/octet-stream',
       compression_type=CompressionTypes.AUTO):
+    # type: (...) -> BinaryIO
+
     """Helper functions to open a file in the provided mode.
     """
     compression_type = FileSystem._get_compression_type(path, compression_type)
-    raw_file = open(path, mode)
+    if mode == 'r':
+      mode = 'rb'
+    elif mode == 'w':
+      mode = 'wb'
```

Comment:
I saw this pattern another filesystem so I thought I'd add it here to make it completely obvious that the subsequent cast to `BinaryIO` (from I`O[Any]`) is safe and warranted.